### ISSUE TITLE
How to run react-native e2e example?

### ISSUE THREAD

**Issue by rahulnainwal107** [microsoft/onnxruntime#12700]
There is no sufficient Information about [running the react-native example](https://github.com/microsoft/onnxruntime/tree/main/js/react_native) and also no detailed information about how to uses this library in react-native.

**Comment by baijumeswani** [comment]
@fs-eire @shalvamist Do we have good documentation/tutorials we can share?

**Comment by fs-eire** [comment]
I have a hello-world project here: https://github.com/fs-eire/ort-rn-hello-world

**Comment by rahulnainwal107** [comment]
Hi @fs-eire Thanks for sharing it. I have checked it but this is all about importing ort models in react-native. There is no info about -
- How to convert an image into tensor.
- How to make predictions using ort models.

**Comment by fs-eire** [comment]
- How to convert an image into tensor.

Great question. perhaps we should introduce some utility functions to do this. I think this is a quite common requirement.

- How to make predictions using ort models.

once the question above is resolved, it's easy to feed the input tensor to ort by `const outputs = await session.run(feeds);`. Code examples can be found in the hello-world projects

**Comment by rahulnainwal107** [comment]
Thanks. Surly that will be great to have an utility function for converting an image into tensor.

**Comment by rahulnainwal107** [comment]
Hi @fs-eire, Is there any update for utility function?

**Comment by fs-eire** [comment]
we are recently get started on the work. how do you expect to load the image - is it from a URL, an <img>, a <canvas> or other form?

**Comment by rahulnainwal107** [comment]
Much thanks @fs-eire, I am trying to load a local image and display it using [Image component ](https://reactnative.dev/docs/image).

**Comment by rahulnainwal107** [comment]
Hi @fs-eire, Is there any update for it?

**Comment by YUNQIUGUO** [comment]
Hi @rahulnainwal107,

Regarding loading a local image in react-native application,  I've used ImagePicker in my Expo react native application: https://docs.expo.dev/versions/latest/sdk/imagepicker/#usage

Here's some sample code snippet:
```
export default function AndroidApp({ navigation, route }: MainScreenProps) {
  const [selectedImage, setSelectedImage] = useState<any>(null);
}

async function openImagePickerAsync() {

    const permissionResult = await ImagePicker.requestMediaLibraryPermissionsAsync();

    if (permissionResult.granted === false) {
      alert("Permission to access Camera Roll is Required!");
      return;
    }

    const pickerResult = await ImagePicker.launchImageLibraryAsync({ allowsEditing: true, base64: true });

    if (pickerResult.cancelled === true) {
      return;
    }

    setSelectedImage({
      localUri: pickerResult.uri
    });
 
 };

 </View>
      {
        selectedImage !== null &&
        <ScrollView style={styles.scrollView}>
          <Image
            source={{ uri: selectedImage.localUri }}
            style={styles.thumbnail}
          />
          {
        </ScrollView>}
```

This is based on steps:

- Upload local image to android emulator gallery https://stackoverflow.com/questions/29588124/how-to-add-an-image-to-the-emulator-gallery-in-android-studio or have the image ready in photos folder in your android phone
- Once you have the local image ready, imagepicker will select the picture and in the image source set it to the selected image uri
- In the first ImagePicker api link, it uses a simplier way to load image directly from assets folder. can also take a look.

And regarding your previous question on how to make the image into ort tensor:
- Currently, you will need to enable passing a base64 encoded image string when passing in the jpg/png image from the app.
  If you are using ImagePicker: there's a base64 option allow you to do that: https://docs.expo.dev/versions/latest/sdk/imagepicker/#imagepickerasset 
- Once you have the base64 string, use the below sample code to turn it into bytearray and wrap it to an ort tensor:
```
import { btoa, atob } from 'react-native-quick-base64';

let inputDataArray = base64js.toByteArray(base64ImageStringFromUpload)
let dataLength = inputDataArray.length  
let tensor = new ort.Tensor(inputDataArray, [dataLength])
```
Note that it uses a base64js npm library : https://www.npmjs.com/package/react-native-quick-base64 found more information here.
Also the sample code uses uint8 data type and tensor shape may also needs to get updated according to your model. Would you share which model you will be using? And all code snippets are just for android side as an example. 

Plus, we are working on a sample react-native expo cross platform application with super resolution model, once that sample is ready, would provide more details on how to run an react-native application end-to-end.

**Comment by rahulnainwal107** [comment]
Hi @fs-eire / @YUNQIUGUO, Much thanks for these information. Will try it.

> Would you share which model you will be using? 

I didn’t understand but i am using .ort and .onnx models and using float32 data type.



> Also the sample code uses uint8 data type and tensor shape may also needs to get updated according to your model.

How we can specify data type and tensor shape?

**Comment by YUNQIUGUO** [comment]
> How we can specify data type and tensor shape?

Refer to https://github.com/microsoft/onnxruntime-inference-examples/blob/main/js/api-usage_tensor/tensor-create.js#L15 there are some other good examples there. can take a look.

And I was just curious which specific onnx model you are using for us to have a general idea, it's good to know the model input is float32.

**Comment by skottmckay** [comment]
React Native basic usage is now covered by https://github.com/microsoft/onnxruntime-inference-examples/tree/main/mobile/examples/React_Native/ort-rn-basic-usage

You can also add pre/post processing to the model. e.g. convert image bytes into a tensor. 
https://github.com/microsoft/onnxruntime-extensions/blob/main/onnxruntime_extensions/tools/Example%20usage%20of%20the%20PrePostProcessor.md

This script handles a few common models and can be used as an example: https://github.com/microsoft/onnxruntime-extensions/blob/main/onnxruntime_extensions/tools/add_pre_post_processing_to_model.py

Also some e2e tutorials for different model types. 
https://github.com/microsoft/onnxruntime-extensions/blob/main/tutorials/yolo_e2e.py
https://github.com/microsoft/onnxruntime-extensions/blob/main/tutorials/superresolution_e2e.py
https://github.com/microsoft/onnxruntime-extensions/blob/main/tutorials/bert_e2e.py

**Comment by EssamWisam** [comment]
@skottmckay I followed the steps in the first example app you linked to multiple time and I consistently get an error `Cannot read property install of null` and the source of this is `OnnxruntimeJSHelper.install()`. Any recommendations or should I create another issue for this?
